src/ember-app.js

```diff
@@ -345,42 +299,44 @@ class EmberApp {
     });
 
     let doc = bootOptions.document;
+    let result = new Result(doc, html, fastbootInfo);
 
-    let result = new Result({
-      doc: doc,
-      html: html,
-      fastbootInfo: fastbootInfo,
-    });
-
+    // TODO: Use Promise.race here
     let destroyAppInstanceTimer;
     if (destroyAppInstanceInMs > 0) {
       // start a timer to destroy the appInstance forcefully in the given ms.
       // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
       destroyAppInstanceTimer = setTimeout(function() {
-        if (result._destroyAppInstance()) {
+        if (result._destroy()) {
```

Comment:
This condition will always be false for visits that never completed so we will never set `result.error`

Reply:
Aha! thank you, will fix this (I don't think it will need to be conditional in the end once we race, but in the meantime I'll bring back the true/false).

Reply:
yeah that is why I asked :)

Reply:
OK, fixed this with the latest commit.